checkpointManager.py

```diff
@@ -1,9 +1,98 @@
 from checkpoints import *
-
+from scenarioManager import *
 class CheckpointManager:
-    def __init__(self, checkpoints):
+    def __init__(self, checkpoints,scenarioManager):
         self.checkpoints = checkpoints
+        #to track where we're at, the state of the player within the checkpoints
+        self.currentCheckpoint = checkpoints[0]
+        self.scenarioManager = scenarioManager
     
+    #checkpoints are linear, so this will advance one checkpoint forward. if the checkpoint manager is already at the last checkpoint, call the gameover scenario
     def updateState(self):
```

Comment:
- [ ] Rename to nextCheckpoint
- [ ] Needs to update/return the distance to the next checkpoint in the player class
- [ ] Needs to return the updated money of the player (for ferry / food)
- [ ] Needs to return the amount of food bought